src/Raven.Server/Web/System/DatabasesHandler.cs

```diff
@@ -297,50 +297,25 @@ private Task DbInfo(string dbName)
         private void WriteDatabaseInfo(string databaseName, BlittableJsonReaderObject dbRecordBlittable,
             TransactionOperationContext context, AbstractBlittableJsonTextWriter writer)
         {
+            var nodesTopology = new NodesTopology();
+
             try
             {
                 var online = ServerStore.DatabasesLandlord.DatabasesCache.TryGetValue(databaseName, out Task<DocumentDatabase> dbTask) &&
                              dbTask != null &&
                              dbTask.IsCompleted;
 
-                // Check for exceptions
-                if (dbTask != null && dbTask.IsFaulted)
-                {
-                    var exception = dbTask.Exception.ExtractSingleInnerException();
-                    WriteFaultedDatabaseInfo(databaseName, exception, context, writer);
-                    return;
-                }
-
                 var dbRecord = JsonDeserializationCluster.DatabaseRecord(dbRecordBlittable);
-                var db = online ? dbTask.Result : null;
-
-                var indexingStatus = db?.IndexStore?.Status;
-                if (indexingStatus == null)
-                {
-                    // Looking for disabled indexing flag inside the database settings for offline database status
-                    if (dbRecord.Settings.TryGetValue(RavenConfiguration.GetKey(x => x.Indexing.Disabled), out var val) && 
-                        bool.TryParse(val, out var indexingDisabled) && indexingDisabled)
-                        indexingStatus = IndexRunningStatus.Disabled;
-                }
-
-                var disabled = dbRecord.Disabled;
                 var topology = dbRecord.Topology;
-                var clusterTopology = ServerStore.GetClusterTopology(context);
-                clusterTopology.ReplaceCurrentNodeUrlWithClientRequestedNodeUrlIfNecessary(ServerStore, HttpContext);
-
-                var studioEnvironment = StudioConfiguration.StudioEnvironment.None;
-                if (dbRecord.Studio != null && !dbRecord.Studio.Disabled)
-                {
-                    studioEnvironment = dbRecord.Studio.Environment;
-                }
-                
-                var nodesTopology = new NodesTopology();
 
                 var statuses = ServerStore.GetNodesStatuses();
                 if (topology != null)
                 {
                     nodesTopology.PriorityOrder = topology.PriorityOrder;
-                    
+
+                    var clusterTopology = ServerStore.GetClusterTopology(context);
```

Comment:
This is only gathered if `topology` is not null? Previously there was no relation between `topology` and `clusterTopology` I think?

Reply:
We only use it to `GetUrlFromTag` and we do that only if there is topology